### PR TITLE
feat: remove `-e` and add `--dir` flag to flox init

### PR DESCRIPTION
**Pull request by ysndr**

### CODE REVIEW

**Review by ghudgins — APPROVED** [review]

File: crates/flox/doc/flox-init.md [hunk 1]
```diff
@@ -12,19 +12,22 @@ flox-init - initialize flox expressions for current project.
...
 
-An existing package called `<name>` will raise an error.
+The name of the environment will be the basename of the current directory
+or "default" if the current directory is `$HOME`.
+The `--name` flag can be used to give the environment a custom name.
 
-If `<template>` or `<name>` are unspecified and a controlling TTY is present,
-flox will query them using interactive dialogs.
+By default the environment will be created in the current directory.
+Flox will add a directory `$PWD/.flox`,
```
Comment (mkenigs) [inline crates/flox/doc/flox-init.md:25]:
```suggestion
flox will add a directory `$PWD/.flox`,
```

File: crates/flox/doc/flox-init.md [hunk 2]
```diff
@@ -12,19 +12,22 @@ flox-init - initialize flox expressions for current project.
...
+within which all relevant metadata of the environment will be tracked.
```
Comment (mkenigs) [inline crates/flox/doc/flox-init.md:26]: question nonblocking: should we say something about what should be committed to version control?

File: crates/flox/doc/flox-init.md [hunk 3]
```diff
@@ -12,19 +12,22 @@ flox-init - initialize flox expressions for current project.
...
+The `--dir` flag can be used to create an environment in another location.
 
-In non interactive shells the command will fail without a `<name>` and default
-to a generic builder for `<template>`.
+If an environment already exists in the current directory,
+or the path specified using `--dir`, an error is returned.
```
Comment (mkenigs) [inline crates/flox/doc/flox-init.md:30]:
```suggestion
or the path specified using `--dir` exists, an error is returned.
```

**Review by mkenigs — APPROVED** [review]

File: crates/flox/doc/flox-init.md (hunk 2 above)
Comment (ysndr) [reply crates/flox/doc/flox-init.md:26]: I think we should do that once the layout and files are finalized

**Pull request merged**